### PR TITLE
Support TIFs, class weights, and directory exclusion

### LINKED ISSUE #14: Parse TIF files

### PR DESCRIPTION
**Pull request by mcquin**
* Adds supports for concatenating single-channel TIFs into NPYs for training and evaluation
* Uses median sampling
* Automatically sets class weights
* Option to configure directories to exclude for training and evaluation

Resolves #14

### CODE REVIEW

File: deepometry/commands/command_evaluate.py [hunk 1]
```diff
@@ -34,24 +34,38 @@
...
     "--name",
     default=None,
     help="A unique identifier for referencing this model.",
     type=click.STRING
 )
+@click.option(
+    "--samples",
+    default=None,
+    help="Number of objects to be collected per class label to pool into testing dataset."
+         "This setting is useful to limit certain amount of datapoint to be displayed in unsupervised PCA/t-SNE plots."
```
Comment (mcquin) [inline deepometry/commands/command_evaluate.py]: Needs a leading space.

File: deepometry/commands/command_evaluate.py [hunk 2]
```diff
@@ -34,24 +34,38 @@
...
+         "All numpy arrays will be collected for testing if this flag is omitted.",
```
Comment (mcquin) [inline deepometry/commands/command_evaluate.py]: Needs a leading space.

File: deepometry/commands/command_evaluate.py [hunk 3]
```diff
@@ -34,24 +34,38 @@
     help="Output directory for model checkpoints, metrics, and metadata.",
     type=click.Path(exists=True)
 )
+@click.option(
+    "--exclude",
```
Comment (mcquin) [inline deepometry/commands/command_evaluate.py]: This functionality should be tested.

File: deepometry/commands/command_evaluate.py [hunk 4]
```diff
@@ -90,19 +107,23 @@ def _evaluate(x, y, batch_size, directory, name, verbose):
...
+
     x = numpy.empty((len(pathnames),) + _shape(pathnames[0]), dtype=numpy.uint8)
 
     y = numpy.empty((len(pathnames),), dtype=numpy.uint8)
 
     label_to_index = {label: index for index, label in enumerate(sorted(labels))}
 
     for index, pathname in enumerate(pathnames):
-        label = os.path.split(os.path.dirname(pathname))[-1]
+        if os.path.isfile(pathname):  # in case there is a mixture of directories and files
```
Comment (mcquin) [inline deepometry/commands/command_evaluate.py]: This filtering can be done earlier, in sample.

File: deepometry/commands/command_fit.py [hunk 5]
```diff
@@ -39,6 +38,12 @@
     help="Number of iterations over training data.",
     type=click.INT
 )
+@click.option(
+    "--exclude",
```
Comment (mcquin) [inline deepometry/commands/command_fit.py]: This functionality should be tested.

File: deepometry/commands/command_fit.py [hunk 6]
```diff
@@ -85,19 +90,23 @@ def command(input, batch_size, directory, epochs, name, validation_split, verbos
...
+
     x = numpy.empty((len(pathnames),) + _shape(pathnames[0]), dtype=numpy.uint8)
 
     y = numpy.empty((len(pathnames),), dtype=numpy.uint8)
 
     label_to_index = {label: index for index, label in enumerate(sorted(labels))}
 
     for index, pathname in enumerate(pathnames):
-        label = os.path.split(os.path.dirname(pathname))[-1]
+        if os.path.isfile(pathname):  # in case there is a mixture of directories and files
```
Comment (mcquin) [inline deepometry/commands/command_fit.py]: This filtering can be done earlier, in sample.

File: deepometry/parse.py [hunk 7]
```diff
@@ -1,43 +1,52 @@
...
+SUPPORTED_FORMATS = [".cif", ".tif", ".tiff"]
 
-def parse(pathname, output_directory, size, channels=None):
-    """
-    Convert an .CIF image file to a NumPy array. The javabridge JVM is required:
 
-        import bioformats
-        import deepometry.parse
-        import javabridge
+def parse(paths, output_directory, size, channels=None):
```
Comment (mcquin) [inline deepometry/parse.py]: Needs documentation.

**Comment by mcquin** [comment]
@minh-doan For .TIF data, is the channel information (`Ch0`) prefixed onto the filename, or suffixed? The example notebooks specify prefix, but the implementation is suffix.

**Comment by minh-doan** [comment]
@mcquin it's usually suffixed for most users, or sometimes middle, for e.g. DAPI_137_Ch0.ome.tif
My bad leaving the prefix in example notebooks !

**Comment by mcquin** [comment]
Thanks for the clarification, @minh-doan! :) I've cleaned up the examples a little.

**Pull request merged**